### PR TITLE
[fix](be)the set operation node should accept both nullable and non-nullable data from child node

### PR DESCRIPTION
**Pull request by starocean999**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by morrySnow — DISMISSED** [review]

File: be/src/vec/exec/vset_operation_node.cpp [hunk 1]
```diff
@@ -239,13 +239,22 @@ Status VSetOperationNode<is_intersect>::prepare(RuntimeState* state) {
     _pull_timer = ADD_TIMER(runtime_profile(), "PullTime");
 
     // Prepare result expr lists.
+    vector<bool> nullable_flags;
+    nullable_flags.resize(_child_expr_lists[0].size(), false);
     for (int i = 0; i < _child_expr_lists.size(); ++i) {
+        int j = 0;
+        for (auto ctx : _child_expr_lists[i]) {
```
Comment (HappenLee) [inline be/src/vec/exec/vset_operation_node.cpp:246]: better directly use ```for(int j = 0; j < _child_expr_lists[i].size()```

File: be/src/vec/exec/vset_operation_node.cpp [hunk 2]
```diff
@@ -239,13 +239,22 @@ Status VSetOperationNode<is_intersect>::prepare(RuntimeState* state) {
...
+            nullable_flags[j] = nullable_flags[j] || ctx->root()->is_nullable();
+            ++j;
+        }
         RETURN_IF_ERROR(VExpr::prepare(_child_expr_lists[i], state, child(i)->row_desc()));
     }
-
+    int n = 0;
     for (auto ctx : _child_expr_lists[0]) {
-        _build_not_ignore_null.push_back(ctx->root()->is_nullable());
```
Comment (HappenLee) [inline be/src/vec/exec/vset_operation_node.cpp:247]: save to upper

File: be/src/vec/exec/vset_operation_node.cpp (hunk 1 above)
Comment (HappenLee) [inline be/src/vec/exec/vset_operation_node.cpp:246]: better directly use ```for(int j = 0; j < _child_expr_lists[i].size()```

Comment (HappenLee) [inline be/src/vec/exec/vset_operation_node.cpp:246]: better directly use ```for(int j = 0; j < _child_expr_lists[i].size()```

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**